Origin: upstream, commit:cd3b3822e17416e3c7e16d72751083a2d362c606
Author: Tom Cannon <tom-cannon@playcannon.com>
Description: fix #282246: "other" appearance in time sig props not working.
 This bug occurs when the user chooses an "other" time signature
 property. This fix works by changing the implementation to be closer to
 MuseScore 2. This works by simply only checking to use the numbers if
 the numerator string is empty. This way, if the denominator string is
 empty, the numerator string which will have the custom symbol, will line
 up in the middle. This was how the implementation worked in MuseScore 2.

--- a/libmscore/timesig.cpp
+++ b/libmscore/timesig.cpp
@@ -299,8 +299,14 @@ void TimeSig::layout()
             ds.clear();
             }
       else {
-            ns = toTimeSigString(_numeratorString.isEmpty()   ? QString::number(_sig.numerator())   : _numeratorString);
-            ds = toTimeSigString(_denominatorString.isEmpty() ? QString::number(_sig.denominator()) : _denominatorString);
+            if (_numeratorString.isEmpty()) {
+                  ns = toTimeSigString(_numeratorString.isEmpty()   ? QString::number(_sig.numerator())   : _numeratorString);
+                  ds = toTimeSigString(_denominatorString.isEmpty() ? QString::number(_sig.denominator()) : _denominatorString);
+                  }
+            else {
+                  ns = toTimeSigString(_numeratorString);
+                  ds = toTimeSigString(_denominatorString);
+                  }
 
             ScoreFont* font = score()->scoreFont();
             QSizeF mag(magS() * _scale);
--- a/mscore/timesigproperties.cpp
+++ b/mscore/timesigproperties.cpp
@@ -114,6 +114,10 @@ TimeSigProperties::TimeSigProperties(Tim
                         textButton->setChecked(false);
                         otherButton->setChecked(true);
                         otherCombo->setCurrentIndex(idx);
+
+                        // set the custom text fields to empty
+                        zText->setText(QString());
+                        nText->setText(QString());
                         }
                   }
             idx++;
@@ -134,21 +138,12 @@ TimeSigProperties::TimeSigProperties(Tim
 void TimeSigProperties::accept()
       {
       TimeSigType ts = TimeSigType::NORMAL;
-      if (textButton->isChecked())
+      if (textButton->isChecked() || otherButton->isChecked())
             ts = TimeSigType::NORMAL;
       else if (fourfourButton->isChecked())
             ts = TimeSigType::FOUR_FOUR;
       else if (allaBreveButton->isChecked())
             ts = TimeSigType::ALLA_BREVE;
-      else if (otherButton->isChecked()) {
-            // if other symbol, set as normal text...
-            ts = TimeSigType::NORMAL;
-            ScoreFont* scoreFont = timesig->score()->scoreFont();
-            SymId symId = (SymId)( otherCombo->itemData(otherCombo->currentIndex()).toInt() );
-            // ...and set numerator to font string for symbol and denominator to empty string
-            timesig->setNumeratorString(scoreFont->toString(symId));
-            timesig->setDenominatorString(QString());
-            }
 
       Fraction actual(zActual->value(), nActual->value());
       Fraction nominal(zNominal->value(), nNominal->value());
@@ -160,6 +155,14 @@ void TimeSigProperties::accept()
       if (nText->text() != timesig->denominatorString())
             timesig->setDenominatorString(nText->text());
 
+      if (otherButton->isChecked()) {
+            ScoreFont* scoreFont = timesig->score()->scoreFont();
+            SymId symId = (SymId)( otherCombo->itemData(otherCombo->currentIndex()).toInt() );
+            // ...and set numerator to font string for symbol and denominator to empty string
+            timesig->setNumeratorString(scoreFont->toString(symId));
+            timesig->setDenominatorString(QString());
+            }
+
       Groups g = groups->groups();
       timesig->setGroups(g);
       QDialog::accept();
